test/parallel/test-vm-measure-memory.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const vm = require('vm');
+const {
+  SUMMARY,
+  DETAILED
+} = vm.constants.measureMemory.mode;
+
+common.expectWarning('ExperimentalWarning',
+                     'vm.measureMemory is an experimental feature. ' +
+                     'This feature could change at any time');
+
+// Test measuring memory of the current context
+{
+  vm.measureMemory(undefined)
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({})
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({ mode: SUMMARY })
+    .then((result) => {
+      assert(result instanceof Object);
+    });
+
+  vm.measureMemory({ mode: DETAILED })
+    .then((result) => {
+      assert(result instanceof Object);
+    });
```

Comment:
Some tests that inspect the contents of `result` would be good to have too.

Reply:
I originally did that but on second thought, this might break as we update V8, which seems undesirable?

Reply:
I'd prefer the test to fail and need to be updated if v8 changes the behavior here, so that at the very least we know :-)

Reply:
Added some tests with comments saying those might need to be updated when V8 is updated